expr/funcs.go

```diff
@@ -2,55 +2,16 @@ package expr
 
 import "github.com/raintank/metrictank/api/models"
 
-//go:generate stringer -type=argType
-type argType uint8
-
-// argument types. to let functions describe their inputs and outputs
-// potentially not as strict as reality (e.g. movingAverage windowsize is categorized as a str) that's why we have the extra validation step
-const (
-	series      argType = iota // a single series
-	seriesList                 // a list of series
-	seriesLists                // one or multiple seriesLists
-	integer                    // number without decimals
-	integers                   // one or multiple numbers without decimals
-	float                      // number potentially with decimals
-	str                        // string
-	boolean                    // True or False
-)
-
-type optArg struct {
-	key string
-	val argType
-}
-
 type Func interface {
```

Comment:
this all exists in the package "expr".  "expr.Func" does nothing to convey what this interface is for.
It needs to be called something else.  GraphiteFunc is the most clear.  That is exactly what we are implementing.

If you want to rename the package from "expr" to "graphite" then that would also be fine.